infra/templates/az-isolated-service-single-region/tests/unit/unit_test.go

```diff
@@ -145,5 +158,19 @@ func TestTemplate(t *testing.T) {
 		},
 	}
 
+	// Required because there is a VNET query done by the template that requires a call to Azure CLI
+	azureLogin(t)
 	infratests.RunUnitTests(&testFixture)
 }
+
+func azureLogin(t *testing.T) {
```

Comment:
let's centralize the login step into a single function managed within the azure extension module.